packages/@expo/cli/src/start/server/metro/serializeHtml.ts

```diff
@@ -91,10 +102,10 @@ function htmlFromSerialAssets(
               return '';
             }
             // Mark async chunks as defer so they don't block the page load.
-            // return `<script src="${baseUrl}/${filename}" defer></script>`;
+            // return `<script src="/${combineUrlPath(baseUrl, filename)" defer></script>`;
           }
 
-          return `<script src="${baseUrl}/${filename}" defer></script>`;
+          return `<script src="/${combineUrlPath(baseUrl, filename)}" defer></script>`;
```

Comment:
Adding the root `/` seems incorrect to me, as this would break `baseUrl`'s like `https://cdn.example.app/` base URLs. Unfortunately, the **e2e/__tests__/export/base-path.test.ts** breaks without it.

@EvanBacon do you know the exact behavior we should follow here? E.g. keep the forced-starting `/`, or correct the e2e test and add another test for `https://cdn.example.app/` base urls.